joblib/numpy_pickle_utils.py

```diff
@@ -147,20 +151,42 @@ def _read_fileobject(fileobj, filename, mmap_mode=None):
     """
     # Detect if the fileobj contains compressed data.
     compressor = _detect_compressor(fileobj)
+    if isinstance(fileobj, tuple(_COMPRESSOR_OBJS)):
```

Comment:
It was to be able to be able to manage in the same function the compatibility with old pickles. The problem was that the `load_compatibility` function had to be called from the initial calling function (e.g `load`). The solution is to yield the filename in case of old pickles and yield file objects with new pickles.